app/controllers/concerns/shopify_app/scopes_verification.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module ScopesVerification
+    extend ActiveSupport::Concern
+    include ScopeUtilities
+
+    included do
+      before_action do
+        login_on_scope_changes(current_merchant_scopes, configuration_scopes)
+      end
+    end
+
+    protected
+
+    def login_on_scope_changes(current_merchant_scopes, configuration_scopes)
```

Comment:
Would it make sense not to pass in the scope sets as params here?

Reply:
I think the intent is to allow partners the freedom to specify their own sources for these strings. I don't know of a better way to offer this flexibility, or whether it's the right move.

Reply:
Makes a lot of sense to me! IMHO if they add value we should keep them.

Reply:
Thinking about this, I think it doesn't give them freedom with the current approach. Because the concern already has the `before_action` which calls the `current_merchant_scopes` and `configuration_scopes ` methods. They would have to override those methods in order to customize.